skylib/push.bzl

```diff
@@ -46,6 +46,46 @@ def _get_runfile_path(ctx, f):
 def _impl(ctx):
     """Core implementation of container_push."""
 
+    if K8sPushInfo in ctx.attr.image:
+        # the image was already pushed, just rename if needed. Ignore registry and repository parameters
+        kpi = ctx.attr.image[K8sPushInfo]
+        actual_pusher = ctx.attr.image[DefaultInfo].files_to_run.executable.short_path if ctx.attr.image[DefaultInfo].files_to_run.executable else ""
+        ctx.actions.write(
+            content = "#!/bin/bash\n{actual_pusher}\n".format(
+                actual_pusher = actual_pusher,
+            ),
+            output = ctx.outputs.executable,
+            is_executable = True,
+        )
+
+        runfiles = ctx.runfiles(files = []).merge(ctx.attr.image[DefaultInfo].default_runfiles)
+
+        ctx.actions.run_shell(
+            tools = [kpi.digestfile],
+            outputs = [ctx.outputs.digest],
+            command = "cp -f \"$1\" \"$2\"",
+            arguments = [kpi.digestfile.path, ctx.outputs.digest.path],
+            mnemonic = "CopyFile",
+            progress_message = "Copying files",
+            use_default_shell_env = True,
+            execution_requirements = {"no-remote": "1", "no-cache": "1"},
```

Comment:
```suggestion
            execution_requirements = {"no-remote": "1", "no-cache": "1"},  # It is is more efficient to locally copy file (which may come from the cache) rather than talk to remote cache. See https://github.com/aspect-build/bazel-lib/blob/e9b66b5e0a11946853c20ad4781abc077ba2a9fe/lib/private/copy_common.bzl#L4 for the 
```